ecc/bls12-377/fr/iop/quotient.go

```diff
@@ -0,0 +1,176 @@
+// Copyright 2020 ConsenSys Software Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Code generated by consensys/gnark-crypto DO NOT EDIT
+
+package iop
+
+import (
+	"math/big"
+	"math/bits"
+
+	"github.com/consensys/gnark-crypto/ecc"
+	"github.com/consensys/gnark-crypto/ecc/bls12-377/fr"
+	"github.com/consensys/gnark-crypto/ecc/bls12-377/fr/fft"
+)
+
+// ComputeQuotient returns h(f₁,..,fₙ)/Xⁿ-1 where n=len(f_i).
+func ComputeQuotient(entries []*Polynomial, h MultivariatePolynomial, expectedForm Form, domains [2]*fft.Domain) (Polynomial, error) {
+
+	var quotientLagrangeCosetBitReverse Polynomial
+
+	// check that the sizes are consistant
+	err := checkSize(entries)
+	if err != nil {
+		return quotientLagrangeCosetBitReverse, nil
+	}
+
```

Comment:
add a check that domains[...] are correct; for example that domains[n].Cardinality != 0 ?